python/hail/linalg/blockmatrix.py

```diff
@@ -707,8 +925,41 @@ def entries(self):
         :class:`.Table`
             Table with a row for each entry.
         """
-        hc = Env.hc()
-        return Table(self._jbm.entriesTable(hc._jhc))
+        return Table(self._jbm.entriesTable(Env.hc()._jhc))
 
 
 block_matrix_type.set(BlockMatrix)
+
+
+def _ndarray_as_matrix_float64(nd):
+    if nd.ndim == 1:
+        if nd.shape[0] == 0:
```

Comment:
Nothing wrong with 0-dimensional arrays:

```
>>> import numpy as np
>>> x = np.zeros((0, 5))
>>> y = np.zeros((5, 9))
>>> x.dot(y)
array([], shape=(0, 9), dtype=float64)
```

Reply:
Good point. I moved the check out of this function (now called  `_ndarray_as_2d_float64`). The error will be at the level of incompatible dimensions for operation.

I still require that the ndarray be explicitly 2d with non-zero entries for `from_numpy` because I think it's good that users have to shape a 1d ndarray to a 2d ndarray before a potentially costly operation where they meant to build a matrix rather than an (inefficient) block vector.

For now BlockMatrix still requires non-zero dimensions, but it looks like Breeze 0.12 may have better support for zero dimensions, so it's something we could consider changing if useful.